app/src/main/java/org/groundplatform/android/repository/LocationOfInterestRepository.kt

```diff
@@ -171,4 +171,28 @@ constructor(
     getValidLois(survey)
       .map { lois -> lois.filter { bounds.contains(it.geometry) } }
       .distinctUntilChanged()
+
+  /**
+   * Deletes a LOI by creating a DELETE mutation, applying it to the local db, and scheduling a task
+   * for remote sync. In free-form jobs, this will also delete associated submissions.
+   *
+   * @param loi The LocationOfInterest to delete
+   */
+  suspend fun deleteLoi(loi: LocationOfInterest) {
```

Comment:
Can we add following validations to this method?

1. Current logged in user is the owner of the LOI or a survey organizer.
2. LOI's "isPredefined" property should be false.